pkg/deploy/spindeploy/transformer/defaults.go

```diff
@@ -0,0 +1,96 @@
+package transformer
+
+import (
+	"context"
+	"fmt"
+	"github.com/armory/spinnaker-operator/pkg/apis/spinnaker/interfaces"
+	"github.com/armory/spinnaker-operator/pkg/generated"
+	"github.com/armory/spinnaker-operator/pkg/util"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// defaultsTransformer inserts default values to *-local profile to each service
+type defaultsTransformer struct {
+	svc    interfaces.SpinnakerService
+	log    logr.Logger
+	client client.Client
+}
+
+type defaultsTransformerGenerator struct{}
+
+func (g *defaultsTransformerGenerator) GetName() string {
+	return "Defaults"
+}
+
+func (a *defaultsTransformerGenerator) NewTransformer(
+	svc interfaces.SpinnakerService,
+	client client.Client,
+	log logr.Logger) (Transformer, error) {
+	return &defaultsTransformer{svc: svc, log: log, client: client}, nil
+}
+
+func (a *defaultsTransformer) TransformConfig(ctx context.Context) error {
+	config := a.svc.GetSpinnakerConfig()
+	for profileName, _ := range util.SpinnakerServices {
+		p := a.assertProfile(config, profileName)
+		err := a.setArchaiusDefaults(p, profileName)
+		if err != nil {
+			return fmt.Errorf("found error while handling profile %s: %e", profileName, err)
+		}
+	}
+	return nil
+}
+
+func (a *defaultsTransformer) setArchaiusDefaults(profile interfaces.FreeForm, profileName string) error {
+	if !isJavaService(profileName) {
```

Comment:
See comment about having a method in utils returning all java service names, we can iterate those for adding profiles since the beginning and remove this check.